ksa_compliance/public/js/branch.js

```diff
@@ -0,0 +1,50 @@
+frappe.ui.form.on("Branch", {
+    setup: function (frm) {
+        frm.set_df_property('custom_branch_ids', 'cannot_delete_rows', 1);
+        frm.set_df_property('custom_branch_ids', 'cannot_add_rows', 1);
+    },
+    refresh: async function (frm) {
+        await filter_company_address(frm)
```

Comment:
It's better to call synchronous code before asynchronous code here, otherwise there may be visible delay before the other IDs are added